app-modules/ai/src/Actions/SendMessage.php

```diff
@@ -71,11 +71,13 @@ public function __invoke(AiThread $thread, string $content): Closure
 
         $aiService->ensureAssistantAndThreadExists($thread);
 
-        return $aiService->sendMessage($message, function (AiMessage $response) use ($thread) {
-            $response->thread()->associate($thread);
-            $response->save();
+        return $aiService
+            ->withFiles($data['files'])
```

Comment:
This makes the AI service stateful, since it stores the files in a property. The AI service is a singleton and can be shared across calls, and this means that in the future if the usage of the service changes it could leak information between requests. I think it would be better to introduce a new parameter to pass the files into `sendMessage()`, probably as the second parameter after the content and before the callback. I think you should introduce a named parameter to the callback argument to provide some clarity.

Reply:
This is what I had originally done, and it does make sense, but it also then implies files are supported or potentially supported by any service, which is not true. I probably should not have even put this method/property on this class, but on the `OpenAiGpt4oService` since it is the only service currently supporting uploads. I'll change it back for now, but think we should strive for a more granular approach in the future, even with files being fundamentally tied to sending messages.